gettsim/generic_functions.py

```diff
@@ -26,9 +26,10 @@ def fill_intercepts_at_lower_thresholds(
 
     intercepts_at_lower_thresholds = np.zeros(upper_thresholds.shape) * np.nan
     intercepts_at_lower_thresholds[0] = intercept_at_lowest_threshold
-    for i, up_thr in enumerate(upper_thresholds[1:]):
+    i = 1
+    for i, up_thr in enumerate(np.append(0, upper_thresholds[0:-1])):
```

Comment:
This is a bit ugly. What is it supposed to do? 

Things like this should not be hard-coded. If we make too few assumptions somewhere else, we might need to add an explicit value for the domain of the function (>=0, ...  ?)